src/main/java/org/terasology/blockNetwork/traversal/TraversalResult.java

```diff
@@ -18,10 +18,33 @@
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
 
+/**
+ * The result of a node traversal performed on a graph. It contains a flag to stop the traversal, a predicate for selecting consequent 
+ * nodes to be traversed, the returned result of the traversal, and the value propagated through the traversal.
+ * 
+ * @param <T> Type of nodes in the graph.
+ * @param <U> Return value type from the traversal.
+ * @param <V> Value passed from node visited into nodes that are outgoing from that node.
+ */
 public final class TraversalResult<T, U, V> {
+    /**
+     * A flag specifying whether to stop the traversal or not.
```

Comment:
can you change these to single comment lines //

Reply:
Agree with Michael on this one, having simple comments seems better to me.